lib/webdriveragent.js

```diff
@@ -80,12 +110,15 @@ class WebDriverAgent {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
+      this.setupNoSessionProxy();
       this.setupProxy(sessionId);
       return this.webDriverAgentUrl;
     }
 
     log.info('Launching WebDriverAgent on the device');
 
+    this.setupNoSessionProxy();
```

Comment:
we use _this.url_ instance variable inside setupNoSessionProxy method and it is mandatory, that this variable is properly set before we call it.